pkg/build/generator/generator.go

```diff
@@ -211,6 +211,9 @@ func (g *BuildGenerator) checkLastVersion(bc *buildapi.BuildConfig, lastVersion
 // updateImageTriggers sets the LastTriggeredImageID on all the ImageChangeTriggers on the BuildConfig and
 // updates the From reference of the strategy if the strategy uses an ImageStream or ImageStreamTag reference
 func (g *BuildGenerator) updateImageTriggers(ctx kapi.Context, bc *buildapi.BuildConfig, from, triggeredBy *kapi.ObjectReference) error {
+	if triggeredBy == nil {
```

Comment:
@rhcarvalho we need to understand why we `triggeredBy` is nil. Panic is not something we should guard against, nor the problem here. Panic just indicates you that this function was invoked in a wrong way. Hiding the panic with checking for nil does not fix the original problem, which is "why this function was called with triggeredBy == nil". Find the reason why and add the if to where this function is invoked.